velox/functions/lib/aggregates/SumAggregateBase.h

```diff
@@ -185,19 +185,10 @@ class DecimalSumAggregate
 
   virtual int128_t computeFinalValue(
       functions::aggregate::LongDecimalWithOverflowState* accumulator) final {
-    // Value is valid if the conditions below are true.
-    int128_t sum = accumulator->sum;
-    if ((accumulator->overflow == 1 && accumulator->sum < 0) ||
-        (accumulator->overflow == -1 && accumulator->sum > 0)) {
-      sum = static_cast<int128_t>(
-          DecimalUtil::kOverflowMultiplier * accumulator->overflow +
-          accumulator->sum);
-    } else {
-      VELOX_CHECK(accumulator->overflow == 0, "Decimal overflow");
-    }
-
-    DecimalUtil::valueInRange(sum);
-    return sum;
+    auto sum = computeValidSum(accumulator->sum, accumulator->overflow);
```

Comment:
Assuming this logic is Spark-specific, can we move it to functions/sparksql?

Reply:
It is not only Spark that requires this logic, but Presto also needs it.

Reply:
@liujiayi771 I see. Would you then update PR title and description to explain that this logic is generic and applies to both Presto and Spark. BTW, should there be some changes in the PR for the Presto logic, no?

Reply:
Presto directly uses the `computeFinalValue` function in `functions/lib/aggregates/SumAggregateBase.h`. Therefore, modifying the call in `SumAggregateBase.h` should be sufficient.